### PR TITLE
Retry failed posts from post list

### LINKED ISSUE #6407: Async media: Support retrying posts with failed media
Currently, a post that has failed media will show up in the post list like this:

post-list-6-media-error

To retry the post upload, we need to edit the post and either retry the individual media items, or remove them/replace them.

Depending on the type of media error, it would be great if we offered the Retry option from the post list (and eventually the error notification).

We already do this when the error is related to the post itself, and there weren't any failed media:

post-list-5-post-error-retry

Some implementation work is required to actually support retrying post uploads with failed media. Once that's in place, we could enable retries for cases we can reasonably expect the retry to succeed (e.g. connection errors), and work on more automated fixes (maybe offering to resize the failed media automatically if the error was a file limit type).

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #6407 

Unhides and implements the `Retry` button in Posts List, by making use of the implementation of `UploadService`.

screen shot 2017-10-25 at 7 36 04 pm

To test:
1. start a draft
2. include several media items in it
3. as the media started uploading, exit the editor
4. see the progress notification show something like “1 post, xxxx media files”
5. turn airplane mode ON
6. observe the error is shown in the Posts list, in red, and the `RETRY` button becomes visible on that Post item, along with the snackbar and corresponding notification.
7. turn airplane mode OFF
8. once connectivity is recovered, tap on RETRY
9. see the progress notification appear, the error disappears from the posts list, and the progress is shown.

cc @nbradbury

### CODE REVIEW

**Comment by nbradbury** [comment]
If you run this on a device with a smaller screen, the retry button wraps to the second set of buttons (so it doesn't show up until you tap "More"). Should it be in the first set instead?

screenshot_1509042459

File: WordPress/src/main/java/org/wordpress/android/widgets/PostListButton.java [hunk 1]
```diff
@@ -139,9 +140,18 @@ public static int getButtonIconResId(int buttonType) {
...
                 return R.drawable.ic_chevron_left_blue_wordpress_18dp;
             case BUTTON_RETRY:
-                return R.drawable.ic_refresh_blue_wordpress_18dp;
+                return R.drawable.ic_refresh_red_18dp;
             default:
                 return 0;
         }
     }
+
+    public static int getTextColorResId(int buttonType) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/widgets/PostListButton.java]: Minor tip: I've started using the `@StringRes`, `@DrawableRes` and `@ColorRes` annotations for these type of routines. ie: `public static @ColorRes int getTextColorResId(int buttonType)`

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1225,8 +1225,8 @@
     <string name="error_media_load">Unable to load media</string>
     <string name="error_media_save">Unable to save media</string>
     <string name="error_media_canceled">Uploading media were canceled</string>
-    <string name="error_media_recover_post">We were unable to upload this post\'s media. Please edit the post to try again.</string>
-    <string name="error_media_recover_page">We were unable to upload this page\'s media. Please edit the page to try again.</string>
+    <string name="error_media_recover_post">We were unable to upload this post\'s media. Please tap retry or edit the post to try again.</string>
```
Comment (nbradbury) [inline WordPress/src/main/res/values/strings.xml]: This is really wordy, plus it uses the pronoun "we" which Google [recommends avoiding](https://material.io/guidelines/style/writing.html#writing-language). The user should be able to figure out they can edit or retry without us telling them, so how about simply, "Unable to upload this post\'s media?"

File: WordPress/src/main/java/org/wordpress/android/widgets/PostListButton.java (hunk 1 above)
Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/widgets/PostListButton.java]: Niiice! Added annotations in dd2caaf

File: WordPress/src/main/res/values/strings.xml (hunk 2 above)
Comment (mzorz) [reply WordPress/src/main/res/values/strings.xml]: I'm amazed about how you are so much into details. Thanks! Made the change as suggested in 8cba571

**Comment by mzorz** [comment]
> If you run this on a device with a smaller screen, the retry button wraps to the second set of buttons (so it doesn't show up until you tap "More"). Should it be in the first set instead?

I did this change in 74fc93a, but could you double check please? not sure if it's going to work correctly with the conditionals there are for showing `publish` or `stats` button

**Comment by nbradbury** [comment]
> I did this change in 74fc93a, but could you double check please?

There are quite a few problems here, such as the Publish button disappearing after you hit `More >` and then `< Back`. This is a pretty thorny bit of code, so I think we should revert this change and file a separate issue for it.

screenshot_1509054769

**Comment by mzorz** [comment]
> This is a pretty thorny bit of code, so I think we should revert this change and file a separate issue for it

Sounds good, reverted in 0c1b740 and opened new issue here, I'll get on it ASAP (i.e. before anything goes to `develop`)
https://github.com/wordpress-mobile/WordPress-Android/issues/6793

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**